### PR TITLE
[VL] Support regexp_replace function with position argument

### PR DESCRIPTION
**Pull request by PHILO-HE**
## What changes were proposed in this pull request?
In the main code, gluten will make regexp_replace fall back if its last `pos` arg is specified by user with a non-default value.  We note upstream velox has a fix for this function. So we are removing such fallback for velox backend.

This PR depends on a velox PR which fixes newly found issues: https://github.com/facebookincubator/velox/pull/8387

## How was this patch tested?

Existing spark UTs.

### CODE REVIEW

**Comment by marin-ma** [comment]
Could you update the PR description? Thanks!

**Comment by FelixYBW** [comment]
@PHILO-HE Sorry, I just synced with @codyschierbeck. regexp_replace in Velox isn't ready yet. Let's wait until his PR is merged

**Comment by PHILO-HE** [comment]
Let's wait for upstream velox pr https://github.com/facebookincubator/velox/pull/8333.

**Comment by PHILO-HE** [comment]
As https://github.com/facebookincubator/velox/pull/8387 has been merged, we can land this pr. @marin-ma, do you have any other comment? Thanks!

**Review by marin-ma — APPROVED** [review]
Thanks

**Pull request merged**
